main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -463,6 +474,27 @@ object Unification {
       Ok((s, econstrs, renv, lenv, purifiedEff))
     }
 
+  /**
+    * Purifies the given tvar in the type and unbinds variables in the substitution that reside in a level
+    * that is deeper than the current level.
+    */
+  def purifyEffAndUnbind(tvar: Type.Var, eff: Type)(implicit level: Level, flix: Flix): InferMonad[Type] = {
```

Comment:
Maybe move to Typer?

Reply:
The standard seems to be that anything involving the InferMonad is kept in Unification.scala.